smartsim/settings/dragonRunSettings.py

```diff
@@ -0,0 +1,208 @@
+        if not all(isinstance(host, str) for host in host_list):
+            raise TypeError("host_list argument must be list of strings")
+        self.run_args["exclude"] = ",".join(host_list)
+
+    def set_cpus_per_task(self, cpus_per_task: int) -> None:
+        """Set the number of cpus to use per task
+
+        :param num_cpus: number of cpus to use per task
+        :type num_cpus: int
+        """
+        self.run_args["cpus-per-task"] = int(cpus_per_task)
+
+    def set_tasks(self, tasks: int) -> None:
+        """Set the number of tasks for this job
+
+        :param tasks: number of tasks
+        :type tasks: int
+        """
+        self.run_args["ntasks"] = int(tasks)
+
+    def set_tasks_per_node(self, tasks_per_node: int) -> None:
+        """Set the number of tasks for this job
+
+        :param tasks_per_node: number of tasks per node
+        :type tasks_per_node: int
+        """
+        self.run_args["ntasks-per-node"] = int(tasks_per_node)
+
+    def set_memory_per_node(self, memory_per_node: int) -> None:
+        """Specify the real memory required per node
+
+        :param memory_per_node: Amount of memory per node in megabytes
+        :type memory_per_node: int
+        """
+        self.run_args["mem"] = f"{int(memory_per_node)}M"
+
+    def set_verbose_launch(self, verbose: bool) -> None:
+        """Set the job to run in verbose mode
+
+        This sets ``--verbose``
+
+        :param verbose: Whether the job should be run verbosely
+        :type verbose: bool
+        """
+        if verbose:
+            self.run_args["verbose"] = None
+        else:
+            self.run_args.pop("verbose", None)
+
+    @staticmethod
+    def _fmt_walltime(hours: int, minutes: int, seconds: int) -> str:
+        """Convert hours, minutes, and seconds into valid walltime format
+
+        Converts time to format HH:MM:SS
+
+        :param hours: number of hours to run job
+        :type hours: int
+        :param minutes: number of minutes to run job
+        :type minutes: int
+        :param seconds: number of seconds to run job
+        :type seconds: int
+        :returns: Formatted walltime
+        :rtype
+        """
+        delta = datetime.timedelta(hours=hours, minutes=minutes, seconds=seconds)
+        fmt_str = str(delta)
+        if delta.seconds // 3600 < 10:
+            fmt_str = "0" + fmt_str
+        return fmt_str
+
+    def set_walltime(self, walltime: str) -> None:
+        """Set the walltime of the job
+
+        format = "HH:MM:SS"
+
+        :param walltime: wall time
+        :type walltime: str
+        """
+        self.run_args["time"] = str(walltime)
+
+    def format_env_vars(self) -> t.List[str]:
+        """Build bash compatible environment variable string for Slurm
+
+        :returns: the formatted string of environment variables
+        :rtype: list[str]
+        """
+        return [f"{k}={v}" for k, v in self.env_vars.items()]
```

Comment:
description is not consistent with behavior. it doesn't "build a formatted string" - it builds a list of strings. If we turn around and do `",".join(obj.format_env_vars)`, why not do it here and not return a list?

Reply:
Removed, but I guess your comment applies to `SrunSettings`.